### ISSUE TITLE
Odd collision detection failure in level 2

### ISSUE THREAD

**Issue by derekfountain** [derekfountain/zxwonkyonekey#16]
There's a consistent error in level 2 where he enters the wall on the left side. I've played through the whole game and this is the only place he does it.

**Comment by derekfountain** [comment]
![screenshot_20190118_174726](https://user-images.githubusercontent.com/6505535/51403785-686c4780-1b49-11e9-933c-e4579774995c.png)

**Comment by derekfountain** [comment]
A state save from the bug reproduction showed nothing unusual. The whole game state is exactly as expected. So I went and looked at the collision code which deals with collisions when he's facing left and in the flat, middle part of the jump. Oh look:

```
     case LEFT_FLAT:
       /* Check if he's about to bang his face */
      check_x = x+1;
```
The sprite x,y coord is always the top left pixel of the sprite. He's facing left in this case, so the collision detection needs to check whether he's about to bang his nose, which means checking the pixel he's about to move into. i.e. the current X location minus one. Not plus one!

That bug's been there since I wrote the collision code, which must be the best part of a year ago now. This is the only place in the whole game where it shows.

There really ought to be a way of testing for this sort of thing, but I can't think how to do it on such limited hardware.
